todo_app/app.py

```diff
@@ -1,40 +1,86 @@
-from flask import Flask, render_template, redirect, url_for, request
-
-from todo_app.ViewModel import ViewModel
-from todo_app.data.Trello import trello
-from todo_app.flask_config import Config
-
+import os
+from flask import Flask, render_template, request, redirect
+from todo_app.dbconnection_manager import *
+from todo_app.view import ViewModel
+from todo_app.models.card import Card
 
 def create_app():
     app = Flask(__name__)
-    app.config.from_object(Config)
-    trello.init()
+    app.config.from_object('todo_app.flask_config.Config')
+
+    #error handling for 404
+    @app.errorhandler(404)
+    def not_found(e):
+        return render_template("error.html", error='not found!')
 
-    @app.route('/')
-    def index():
-        item_view_model = ViewModel(
-            trello.get_list_items(), trello.get_current_sort_order())
+    #default
+    @app.route('/', methods=['GET'])
+    def get_index():
+        cardslist = []
+        mongo = DB_Connection_Manager()
+        items = mongo.get_AllItems()
+
+        for item in items:
+            cardslist.append(Card(item))
+
+        item_view_model = ViewModel(cardslist)
         return render_template('index.html', view_model=item_view_model)
 
+
+    # new task
+    @app.route('/new', methods=['GET'])
+    def getnew_post():
+        return render_template('new_task.html')
+
     @app.route('/', methods=['POST'])
-    def add_item():
-        app.logger.info('Adding new item on a card')
-        trello.add_new_item(request.form['title'])
-        return redirect('/')
-
-    @app.route('/actions/<action>/<id>')
-    def move_item(action, id):
-        app.logger.info('Moving an item from ')
-        if action in ['start', 'doing']:
-            trello.update_to_inprogress(id)
-        elif action == 'done':
-            trello.update_to_done(id)
-
-        return redirect('/')
-
-    @app.route('/sortby/<sortby>')
-    def sort_by(sortby):  # pylint:disable=unused-variable
-        trello.set_current_sort_order(sortby)
-        return redirect('/')
+    def post_index():
+        mongo = DB_Connection_Manager()
+        response = mongo.create_task(
+            name = request.form['title'],
+            due = request.form['duedate'],
+            desc = request.form['descarea']
+        )
+        
+        if (str(response) != ""):
+            return redirect('/')
+        else:
+            return render_template("error.html",error="Task creation failed ")
+
+    # edit task
+    @app.route('/edit/<id>', methods=['GET'])
+    def get_edit(id):
+        mongo = DB_Connection_Manager()
+        item = mongo.get_task(id=id)
+        if (str(item) != ""):
+            item_info = Card(item)
```

Comment:
When you have a class like Card to represent an object stored in the DB, it's usually nicest to initialise it ASAP, so inside `get_task` rather than here. Is there any reason you'd ever want the database dictionary instead of a `Card`?